### PR TITLE
Make Crushing Leap deal damage

### LINKED ISSUE #53511: Crushing Leap doesn't deal damage
### Describe the bug

The Crushing Leap spell from the Crushing Leap mutation doesn't deal damage when either aiming at an enemy or past one, instead you land in front of them with no other effect.

### Steps to reproduce

1. Make a character with the Crushing Leap mutation.
2. Cast the Crushing Leap ability, either aimed directly on or past an enemy.
3. Wish that frogs had tear ducts so you could cry.

### Expected behavior

I expected the Crushing Leap to Crush when Leaping.

### Screenshots

image
image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 8b7a449 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    Mythical Martial Arts [MMA],
    Stats Through Kills [stats_through_kills],
    SpeedyDex [speedydex]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Bugfixes "Crushing Leap now deals damage"

#### Purpose of change
Make Crushing Leap deal damage.

#### Describe the solution
Adds a min and max aoe field, as described in the issue for a fix.

#### Describe alternatives you've considered
None

#### Testing
Crushing leap deals damage

#### Additional context
 Closes #53511

### CODE REVIEW

**Comment by Maleclypse** [comment]
I've kicked the test but I have no clue why the PR labeler is failing from looking at the summary

**Comment by GuardianDll** [comment]
I did test, but, as you can see in the comment, the leap effect deal damage pretty weird. Thats why i decided to find some better solution (that no one propose)
#56351may fix it, but im not sure

**Comment by MNG-cataclysm** [comment]
@Maleclypse Could the PR work, or is it a fatal error?

**Comment by GuardianDll** [comment]
Can you also add the same AoE to spell_avian_leap spell? Thank you

**Comment by MNG-cataclysm** [comment]
@GuardianDll I added the area of effect to avian leap.

**Comment by GuardianDll** [comment]
Thats.. Not how you suppose to resolve this merge conflict
Instead of accepting the changes you revert them to unfixed state

File: data/json/mutations/mutation_spells.json [hunk 1]
```diff
@@ -101,6 +101,8 @@
     "damage_type": "bash",
     "min_range": 8,
     "max_range": 8,
+    "min_aoe": 5,
```
Comment (Maleclypse) [inline data/json/mutations/mutation_spells.json:104]:
```suggestion
    "min_aoe": 2,
```

File: data/json/mutations/mutation_spells.json [hunk 2]
```diff
@@ -101,6 +101,8 @@
...
+    "max_aoe": 5,
```
Comment (Maleclypse) [inline data/json/mutations/mutation_spells.json:105]:
```suggestion
    "max_aoe": 2,
```

File: data/json/mutations/mutation_spells.json [hunk 3]
```diff
@@ -125,6 +127,8 @@
     "damage_type": "cut",
     "min_range": 2,
     "max_range": 3,
+    "min_aoe": 5,
```
Comment (Maleclypse) [inline data/json/mutations/mutation_spells.json:130]:
```suggestion
    "min_aoe": 2,
```

File: data/json/mutations/mutation_spells.json [hunk 4]
```diff
@@ -125,6 +127,8 @@
...
+    "max_aoe": 5,
```
Comment (Maleclypse) [inline data/json/mutations/mutation_spells.json:131]:
```suggestion
    "max_aoe": 2,
```

**Comment by Maleclypse** [comment]
> Thats.. Not how you need to resolve this merge conflict Instead of accepting the changes you revert them to unfixed state

Can you double check the changes I've made?   Aoe of 2 should make it so it only hits what you land directly next to correct? Or did they change aoe of 1 to now be 1 square on each side of the target zone?

**Comment by GuardianDll** [comment]
Just tested it, and leap works even worse than i thought
instead of dealing damage for targets **after the jump** (i dont really understand how AoE field interacts with leap, but it seems really smart to work in that way) it deals damage to the target **before the jump**. its hard to explain, but if our character jump on a distance of 1, and aoe is 6, it means it can jump on distance 1, and deal damage to 5 (6 aoe minus 1) targets. But if it tries to jump on distance 6, it wont be able to deal any damage at all
and if there would be an aoe 2, it means it can jump on distance 1 and attack 1 monster ahead, or jump on distance 6 and deal no damage at all
tldr: it doesnt work. id prefer to close this pr at all, and tomorrow ill redo the damage leap spells to be two action spell - first is the leap, and second is subspell, that will deal some damage around the character after the jump.

**Comment by Maleclypse** [comment]
> Just tested it, and leap works even worse than i thought
> instead of dealing damage for targets **after the jump** (i dont really understand how AoE field interacts with leap, but it seems really smart to work in that way) it deals damage to the target **before the jump**. its hard to explain, but if our character jump on a distance of 1, and aoe is 6, it means it can jump on distance 1, and deal damage to 5 (6 aoe minus 1) targets. But if it tries to jump on distance 6, it wont be able to deal any damage at all
> and if there would be an aoe 2, it means it can jump on distance 1 and attack 1 monster ahead (2 aoe minus 1), or jump on distance 6 and deal no damage at all
> tldr: it doesnt work. id prefer to close this pr at all, and tomorrow ill redo the damage leap spells to be two action spell - first is the leap, and second is subspell, that will deal some damage around the character after the jump.

That sounds like it will work better. Thanks for that!

**Comment by GuardianDll** [comment]
Short answer: leap effect is awful and must be burned in fire
long answer: leap spell has some extremely weird behaviour, which prevent using it in any damage dealing process. i cant even desribe the bug itself, but short demo may explain it for a bit
https://imgur.com/a/ShtFrsY

**Comment by Maleclypse** [comment]
@andrei8l had posted this quick and dirty dif in devcord for some ideas as to how to fix this.

```
diff --git a/src/magic.cpp b/src/magic.cpp
index 091a9f2464..240b9e1d8d 100644
--- a/src/magic.cpp
+++ b/src/magic.cpp
@@ -1528,6 +1528,7 @@ void spell::cast_spell_effect( Creature &source, const tripoint &target ) const
 
 void spell::cast_all_effects( Creature &source, const tripoint &target ) const
 {
+    tripoint_abs_ms const target_abs = get_map().getglobal( target );
     if( has_flag( spell_flag::WONDER ) ) {
         const auto iter = type->additional_spells.begin();
         for( int num_spells = std::abs( damage() ); num_spells > 0; num_spells-- ) {
@@ -1543,26 +1544,29 @@ void spell::cast_all_effects( Creature &source, const tripoint &target ) const
             // if a message is added to the casting spell, it will be sent as well.
             source.add_msg_if_player( sp.message() );
 
+            tripoint const target_local = get_map().getlocal( target_abs );
             if( sp.has_flag( spell_flag::RANDOM_TARGET ) ) {
                 if( const cata::optional<tripoint> new_target = sp.random_valid_target( source,
-                        _self ? source.pos() : target ) ) {
+                        _self ? source.pos() : target_local ) ) {
                     sp.cast_all_effects( source, *new_target );
                 }
             } else {
                 if( _self ) {
                     sp.cast_all_effects( source, source.pos() );
                 } else {
-                    sp.cast_all_effects( source, target );
+                    sp.cast_all_effects( source, target_local );
                 }
             }
         }
     } else {
         if( has_flag( spell_flag::EXTRA_EFFECTS_FIRST ) ) {
             cast_extra_spell_effects( source, target );
-            cast_spell_effect( source, target );
+            tripoint const target_local = get_map().getlocal( target_abs );
+            cast_spell_effect( source, target_local );
         } else {
             cast_spell_effect( source, target );
-            cast_extra_spell_effects( source, target );
+            tripoint const target_local = get_map().getlocal( target_abs );
+            cast_extra_spell_effects( source, target_local );
         }
     }
 }
```

**Comment by chrispikula** [comment]
Side note, is this in any way related to:
#60896 ?

**Comment by GuardianDll** [comment]
No, its not related, and its not even fix the issue it closes
